### PR TITLE
2-cat-gifs edit problem copy

### PR DESCRIPTION
**Pull request by hakierka**
### Pull Request Description:
This pull request improves the instructions for Challenge 2 by providing clearer guidance on how to upload a file using the files-from-path module. It addresses the issues with the previous instructions, clarifying the purpose of the module and ensuring users understand where and how to use the files.

### Changes Made:
- Added detailed explanations on why the files-from-path module is needed.
- Clarified the steps to find and specify a cat meme file for uploading.
- Updated the code example to match the improved instructions.

### CODE REVIEW

File: problems/2-cat-gifs/problem.md [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
 
-In IPFS, hashes are referred to as Content Identifiers, or CID for short. In this exercise we're going to transform your data into content addressed data, that you can refer to via a CID, upload it to web3.storage and print out the CID.
+**Why Content Addressing?**
 
-Create a _new_ file for your solution e.g. `ex3.mjs`. Import the w3up library and create the client as before. Install the module `files-from-path` from npm (`npm install files-from-path`) and import it like so:
+On the third web, you hash your content first. This means the hash you generate is unique to the content you created. If you fetch the content by its hash, you can be sure that the content hasn't been tampered with because it "hashes" to the same value. It's like a super-secure way to keep your cat memes pristine and perfect.
+
+**Let's Get Started!**
+
+We're going to transform your cat memes into content-addressed data, upload them to web3.storage, and print out their unique CID (Content Identifier). Once you downloaded the last meme your friend sent you, continue with:
```
Comment (alanshaw) [inline problems/2-cat-gifs/problem.md:17]: > Once you downloaded the last meme your friend sent you

Note sure I understand this bit?

File: problems/2-cat-gifs/problem.md [hunk 2]
```diff
@@ -1,22 +1,46 @@
...
+
+**1. Create a New File**
+Create a _new_ file, like `ex3.mjs`.
+**2. Import w3up and create the client**
+Import the w3up library and create the client as before. 
+**3. Instal the `files-from-path`**
```
Comment (alanshaw) [inline problems/2-cat-gifs/problem.md:23]:
```suggestion
**3. Install the `files-from-path` module**
```

File: problems/2-cat-gifs/problem.md [hunk 3]
```diff
@@ -1,22 +1,46 @@
...
+It  helps read files from your local file system and transforms file paths into a format that can be uploaded to web3.storage. Without this module, you’d be stuck doing all the heavy lifting of file reading and formatting. And who has time for it in this economy?
```
Comment (alanshaw) [inline problems/2-cat-gifs/problem.md:24]:
```suggestion
It helps read files from your local file system and transforms file paths into a format that can be uploaded to web3.storage. Without this module, you’d be stuck doing all the heavy lifting of file reading and formatting - who has time for that in this economy?
```

File: problems/2-cat-gifs/problem.md [hunk 4]
```diff
@@ -1,22 +1,46 @@
...
+Install `npm install files-from-path` and import
```
Comment (alanshaw) [inline problems/2-cat-gifs/problem.md:25]:
```suggestion
Install `npm install files-from-path` and import:
```

File: problems/2-cat-gifs/problem.md [hunk 5]
```diff
@@ -1,22 +1,46 @@
...
 
 ```js
 import { filesFromPaths } from 'files-from-path'
 ```
+**4. Upload the meme**
+Specify the path to the meme:
+```js
+const files = await filesFromPaths(['path-to-your-meme']) 
+```
+and upload it with using function uploadDirectory on client:
```
Comment (alanshaw) [inline problems/2-cat-gifs/problem.md:35]:
```suggestion
and upload it with using the client's `uploadDirectory` function:
```

File: problems/2-cat-gifs/problem.md [hunk 6]
```diff
@@ -1,22 +1,46 @@
...
+```js
+const root = await client.uploadDirectory(files);
+```
 
-Use the `uploadDirectory` function on the client to upload you favourite cat gif from your local file system to web3.storage. You'll receive a CID for the content, use `console.log` to print it out.
+End with printing CID console.log(root.toString());
```
Comment (alanshaw) [inline problems/2-cat-gifs/problem.md:40]:
```suggestion
Finally, print the CID:
```

File: problems/3-delegation/problem.md [hunk 7]
```diff
@@ -4,11 +4,14 @@ Excellent cat gifs. Well done you.
 
 Now, lets talk UCAN.
 
-UCANs are like JWTs with super powers. They’re an essential component in decentralized authorization. They use public key cryptography - so whenever you create a UCAN you _sign it_ with your private key.
+**What are UCANs?**
+UCANs (User Controlled Authorization Networks) are like supercharged passwords. They’re used for secure, decentralized authorization. When you create a UCAN, you sign it with your private key, kind of like signing a document with your unique signature.
 
-The best thing about UCANs is that you can provably delegate "capabilities" that allow others to act on your behalf, without ever sharing your private key.
+**What are UCANs cool?**
```
Comment (alanshaw) [inline problems/3-delegation/problem.md:10]:
```suggestion
**Why are UCANs cool?**
```

File: problems/3-delegation/problem.md [hunk 8]
```diff
@@ -4,11 +4,14 @@ Excellent cat gifs. Well done you.
...
+The best thing about UCANs is that you can securely give permission to others to do things on your behalf without sharing your private key. This is called "delegation."
```
Comment (alanshaw) [inline problems/3-delegation/problem.md:11]:
```suggestion
The best thing about UCANs is that you can securely give permission to others to do things on your behalf without sharing your private key. This is called "delegation".
```

File: problems/3-delegation/problem.md [hunk 9]
```diff
@@ -4,11 +4,14 @@ Excellent cat gifs. Well done you.
...
 
-In this exercise, you're going to delegate an `upload/list` capability to the workshop. Behind the scenes the workshop will create it's own private key, and pass it's public key (it's DID) to you on `stdin`.
+**Your Mission**
+In this exercise, you're going to delegate an upload/list capability to the workshop. The workshop will create its own private key and pass its public key (DID) to you through stdin (standard input).
```
Comment (alanshaw) [inline problems/3-delegation/problem.md:14]:
```suggestion
In this exercise, you're going to delegate an `upload/list` capability to the workshop. The workshop will create its own private key and pass its public key (DID) to you via `stdin` (standard input).
```

File: problems/3-delegation/problem.md [hunk 10]
```diff
@@ -18,20 +21,53 @@ DID => your program => delegation
 
 Then the workshop will be able to **invoke** the capability to list the items _you've_ uploaded to _your_ space. It'll verify it can find that cat gif you uploaded in the previous exercise.
 
-You can read and parse the DID from `stdin` like so:
+One last thing, make sure the delegation remains valid for **more than one hour**.
+
 
+**Steps to complete the task**
+**1. Create new file and read DID from stdin**
```
Comment (alanshaw) [inline problems/3-delegation/problem.md:28]:
```suggestion
**1. Create new file and read a DID from `stdin`**
```

File: problems/3-delegation/problem.md [hunk 11]
```diff
@@ -18,20 +21,53 @@ DID => your program => delegation
...
+Create new file, like `ex4.mjs` and then you can read and parse the DID from 'process.stdin':
```
Comment (alanshaw) [inline problems/3-delegation/problem.md:29]:
```suggestion
Create new file, like `ex4.mjs` and then you can read and parse the DID from `process.stdin`:
```

File: problems/3-delegation/problem.md [hunk 12]
```diff
@@ -18,20 +21,53 @@ DID => your program => delegation
...
 ```js
 import fs from 'node:fs'
 import * as DID from '@ipld/dag-ucan/did'
 
 const data = fs.readFileSync(process.stdin.fd, 'utf-8')
 const principal = DID.parse(data)
 ```
+**2. Create the delegetion**
+Use the client to create a delegation for upload/list to the provided DID:
```
Comment (alanshaw) [inline problems/3-delegation/problem.md:38]:
```suggestion
Use the client to create a delegation for `upload/list` to the provided DID:
```

File: problems/3-delegation/problem.md [hunk 13]
```diff
@@ -18,20 +21,53 @@ DID => your program => delegation
...
+import * as Client from '@web3-storage/w3up-client';
+```
+Create the client and set the expiration time for the delegation:
+```js
+const client = await Client.create();
+const twoHours = 1000 * 60 * 60 * 2; // Two hours in milliseconds
+```
+In the example above, it is set to two hours.
+**3. Generate the delegation**
+Create the delegation for the upload/list capability:
```
Comment (alanshaw) [inline problems/3-delegation/problem.md:49]:
```suggestion
Create the delegation for the `upload/list` capability:
```

**Review by alanshaw — CHANGES_REQUESTED** [review]
We're not using `;` in code samples here (or anywhere for that matter unless strictly necessary), could you please do a pass to remove them?

**Comment by hakierka** [comment]
adding the changes, reopening the pr (closed by accident)

**Comment by hakierka** [comment]
reopen

File: problems/2-cat-gifs/problem.md (hunk 1 above)
Comment (hakierka) [reply problems/2-cat-gifs/problem.md:17]: mind shortcut, changing it

File: problems/2-cat-gifs/problem.md (hunk 2 above)
Comment (hakierka) [reply problems/2-cat-gifs/problem.md:23]: I messed up and merge this pr with a new one, I'll add the changes to the new one!

File: problems/2-cat-gifs/problem.md (hunk 3 above)
Comment (hakierka) [reply problems/2-cat-gifs/problem.md:24]: applied in the new pr

File: problems/2-cat-gifs/problem.md (hunk 4 above)
Comment (hakierka) [reply problems/2-cat-gifs/problem.md:25]: applied in the new pr

**Pull request closed without merging**